core/src/main/java/brooklyn/entity/rebind/transformer/CompoundTransformer.java

```diff
@@ -181,6 +188,18 @@ public Builder changeCatalogItemId(String oldSymbolicName, String newSymbolicNam
                 Preconditions.checkNotNull(newSymbolicName, "new_symbolic_name")+":"+Preconditions.checkNotNull(newVersion, "new_version"));
         }
 
+        /**
+         * Updates all references to a class to a new value
+         * @param oldName the old name of the class
+         * @param newName the new name of the class to be used instead
+         */
+        public Builder renameClass(String oldName, String newName) {
+            return renameClassTag(oldName, newName)
+                .xmlChangeAttribute("//@class[.='" + oldName + "']", newName)
+                .renameType(oldName, newName);
+                //TODO update reference attributes
```

Comment:
When the same object is used at multiple places it is serialized only once and referenced from all other places through the `reference` attribute along with a relative path to the node. When renaming classes the path changes so the value needs to be updated. Not needed for the advanced-networking transforms so I didn't spend time implementing it.